### PR TITLE
Fix: Unicode highlighting being one letter off

### PR DESCRIPTION
**Pull request by AkalUstat**
* [x]  Passed [sanity tests](http://bit.ly/sttm-sanity-tests).
* [x]  Ran `npm test` & fixed newly introduced lint errors.
* [x]  Checked console for errors.
## The Problem
for some reason, with a unicode query, the last highlight index came out as -1, meaning that the last letter did not get highlighted. Verified through React Devtools

## The Fix
to keep the results consistent, I passed the query through the anvaad unicode function with reverse set to true, so that unicode is converted to ascii. Normal queries will pass through unaffected. This way, the results are consistent regardless of english or unicode.

### CODE REVIEW

**Comment by AkalUstat** [comment]
there are a ton of changes but that is due to the package-lock.json when i installed anvaad-js. Otherwise its a 1 line change

File: src/js/util/index.js [hunk 1]
```diff
@@ -317,7 +318,7 @@ export const getHighlightIndices = (baani, query, type) => {
   let mainQuery = query.replace(/"/g, '');
 
   //Handles - search operator
-  mainQuery = mainQuery.replace(/[-][ ,\w,),(]*/g, '');
+  mainQuery = unicode(mainQuery.replace(/[-][ ,\w,),(]*/g, ''), true);
```
Comment (Gauravjeetsingh) [inline src/js/util/index.js:321]: @AkalUstat Why are we converting the search query to unicode here? 

We never used anvaad js directly, API was supposed to use that for us until now.

Comment (Gauravjeetsingh) [reply src/js/util/index.js:321]: Got the answer from your above description. Just thinking if this is the best way to solve it.

What if we don't convert the search query to unicode ever? When switching to unicode, everything else switches, but not the search query.

Comment (AkalUstat) [reply src/js/util/index.js:321]: also i really didnt bother to see what was happening with the highlighted index generator i just plugged this in cause it was the first solution i thought of

Comment (Gauravjeetsingh) [reply src/js/util/index.js:321]: Okay, let me take a look at the highlighted index generator, possibly an error with the regex we use for that. If it's solvable from there, I'd prefer that.

Comment (saintsoldierx) [reply src/js/util/index.js:321]: any update @Gauravjeetsingh ?

Comment (Gauravjeetsingh) [reply src/js/util/index.js:321]: I am not able to reproduce this issue on my laptop, and safari on iPhone X

But the query in autocomplete is never Unicode. So this PR fix doesn't affect that. And as per the screenshot in issue #1097, the highlight is off by one word in autocomplete as well. 

I think it's safe to close this PR **without merging**, but the ticket can stay open. I need to investigate more.

Comment (AkalUstat) [reply src/js/util/index.js:321]: - this is if the user manually enters unicode

Comment (tarunsingh5) [reply src/js/util/index.js:321]: is this still an issue?

**Pull request closed without merging**